common/micc.js

```diff
@@ -16,53 +16,73 @@ function Micc(serverAddress) {
                 "Content-Type": "application/x-www-form-urlencoded"
             },
         })
-        .done(function (data) {
-            console.log('Login success.  Received data:  ', data);
-            bearerToken = data.access_token;
-            if(processResponse) {
-                data.miccServer = miccServerBase;
-                processResponse(data);
-            }
-        });
+            .done(function (data) {
+                console.log('Login success.  Received data:  ', data);
+                bearerToken = data.access_token;
+                if (processResponse) {
+                    data.miccServer = miccServerBase;
+                    processResponse(data);
+                }
+            });
     };
 
     this.getEmployeeStats = function (employeeId, receiveEmployeeStats) {
-        if(!employeeId) {
+        if (!employeeId) {
             console.log('Will not fetch because no employee id given.');
         } else {
             this.getRequest(`employees/${employeeId}/state`, receiveEmployeeStats);
         }
     };
 
     this.getQueueStats = function (queueId, receiveQueueStats) {
-        if(!queueId) {
+        if (!queueId) {
             console.log('Will not fetch because no queue id given.');
         } else {
             this.getRequest(`queues/${queueId}/state`, receiveQueueStats);
         }
     };
 
-    this.postEmployeeConversation = function(employeeId, body, processResponse) {
-        if(!employeeId) {
+    this.getEmployeeBusyReasonCodes = function (employeeId, receiveEmployeeBusyCodes) {
+        if (!employeeId) {
+            console.log('Will not fetch because no employee id given.');
+        } else {
+            this.getRequest(`employees/${employeeId}/busyreasoncodes`, receiveEmployeeBusyCodes);
+        }
+    };
+
+    this.postEmployeeConversation = function (employeeId, body, processResponse) {
+        if (!employeeId) {
             console.log('Will not fetch because no employee id given.');
         } else {
             this.postRequest(`employees/${employeeId}/conversations`, body, processResponse);
         }
     }
 
-    this.postOpenMedia = function(body, processResponse) {
+    this.putEmployeeState = function (employeeId, body, processResponse) {
+        if (!employeeId) {
+            console.log('Will not fetch because no employee id given.');
+        } else {
+            this.putRequest(`employees/${employeeId}/state`, body, processResponse);
+        }
+    }
+
+    this.postOpenMedia = function (body, processResponse) {
         this.postRequest(`openmedia`, body, processResponse);
     }
-    
-    this.getRequest = function(apiSubUrl, processResponse) {
+
+    this.getRequest = function (apiSubUrl, processResponse) {
```

Comment:
There is a space after 'function' keyword for the older code. I'll fix them as well as part of this changeset.